pkgs/development/perl-modules/generic/default.nix

```diff
@@ -36,6 +36,7 @@ toPerlModule(stdenv.mkDerivation (
   {
     name = "perl${perl.version}-${name}";
     builder = ./builder.sh;
+    buildInputs = buildInputs ++ [ perl ];
     nativeBuildInputs = nativeBuildInputs ++ [ (perl.dev or perl) ];
```

Comment:
This effectively puts TWO ```perl``` into ```$PATH```
and it might be indeterministic which one will be used in ```./builder.sh```
https://github.com/NixOS/nixpkgs/blob/d8d2231f3ef792ad04594c921762369459a09a43/pkgs/development/perl-modules/generic/builder.sh#L25

Perhaps ```nativeBuildInputs``` is to be removed then.
i have not tested, just a suspicion.

Reply:
Aren't they the same for native builds? It is definitely needed for cross compilation, so that we can execute perl scripts on the target machine/

Reply:
When they are the same, one of them is enough

Reply:
With strictDeps only the nativeBuildInputs one is in path